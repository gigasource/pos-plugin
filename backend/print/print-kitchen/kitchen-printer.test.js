const { printKitchen, printKitchenCancel } = require("./kitchen-printer");
const { cmsFactory } = require("../../../test-utils");
const mockOrder = require("../../../test-utils/dataMock/mockOrder.json");
const { stringify } = require('schemahandler/utils')
const { prepareKitchenPrinter } = require('./kitchen-printer.prepare.test')

const cms = cmsFactory("testKitchenPrinter");

global.cms = cms;

jest.setTimeout(60000)

describe("Test kitchen printer", function() {
  beforeAll(async () => {
    await cms.init();
    prepareKitchenPrinter(cms)
  });

  it("Test print kitchen", async () => {
    const result = await printKitchen({
      order: mockOrder
    });
    expect(stringify(result)).toMatchSnapshot();
  });

  it('Test print kitchen cancel', async () => {

  })
});
