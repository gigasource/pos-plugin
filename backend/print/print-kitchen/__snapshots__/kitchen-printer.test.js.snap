// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test kitchen printer Test print kitchen 1`] = `
Object {
  "results": Array [
    Array [],
    Array [
      Object {
        "items": Array [
          Object {
            "_id": "ObjectID",
            "activePopupModifierGroup": "ObjectID",
            "attributes": Array [],
            "category": null,
            "choices": Array [],
            "course": 1,
            "groupPrinter": "Kitchen",
            "groupPrinter2": null,
            "id": "",
            "ingredients": Array [],
            "isDivArticle": null,
            "isItemNote": null,
            "isNoPrint": null,
            "modifiers": Array [],
            "name": "PRIME BEEF",
            "option": Object {
              "_id": "ObjectID",
            },
            "originalPrice": 11,
            "price": 11,
            "printed": true,
            "product": "ObjectID",
            "quantity": 1,
            "sent": true,
            "tax": null,
            "tax2": null,
            "taxes": Array [
              null,
              null,
            ],
            "type": null,
          },
          Object {
            "_id": "ObjectID",
            "activePopupModifierGroup": "ObjectID",
            "attributes": Array [],
            "category": null,
            "choices": Array [],
            "course": 1,
            "groupPrinter": "Kitchen",
            "groupPrinter2": null,
            "id": "",
            "ingredients": Array [],
            "isDivArticle": null,
            "isItemNote": null,
            "isNoPrint": null,
            "modifiers": Array [],
            "name": "SINGAPORE STYLE SEAFOOD",
            "option": Object {
              "_id": "ObjectID",
            },
            "originalPrice": 10,
            "price": 10,
            "printed": true,
            "product": "ObjectID",
            "quantity": 1,
            "sent": true,
            "tax": null,
            "tax2": null,
            "taxes": Array [
              null,
              null,
            ],
            "type": null,
          },
          Object {
            "_id": "ObjectID",
            "activePopupModifierGroup": "ObjectID",
            "attributes": Array [],
            "category": null,
            "choices": Array [],
            "course": 1,
            "groupPrinter": "Kitchen",
            "groupPrinter2": null,
            "id": "",
            "ingredients": Array [],
            "isDivArticle": null,
            "isItemNote": null,
            "isNoPrint": null,
            "modifiers": Array [],
            "name": "PRIME BEEF",
            "option": Object {
              "_id": "ObjectID",
            },
            "originalPrice": 11,
            "price": 11,
            "printed": true,
            "product": "ObjectID",
            "quantity": 1,
            "sent": true,
            "tax": null,
            "tax2": null,
            "taxes": Array [
              null,
              null,
            ],
            "type": null,
          },
        ],
        "name": "Kitchen",
        "printer": Object {
          "_id": "ObjectID",
          "escPOS": true,
          "fontSize": 1,
          "hardwares": Array [],
          "includes": Array [],
          "ip": "192.168.10.6",
          "marginTop": 0,
          "merge": true,
          "name": "Kitchen",
          "printerType": "ip",
        },
      },
    ],
  ],
  "success": true,
}
`;
